### PR TITLE
32 bit IntervalYearMonth

### PR DESCRIPTION
**Pull request by wrmsr**
Implemented #5434 before I saw it was assigned :|

### CODE REVIEW

**Comment by martint** [comment]
@haozhun, can you review?

File: presto-main/src/main/java/com/facebook/presto/type/IntervalYearMonthOperators.java [hunk 1]
```diff
@@ -153,7 +153,7 @@ public static boolean between(
     @SqlType(StandardTypes.VARCHAR)
     public static Slice castToSlice(@SqlType(StandardTypes.INTERVAL_YEAR_TO_MONTH) long value)
     {
-        return utf8Slice(IntervalYearMonth.formatMonths(value));
+        return utf8Slice(IntervalYearMonth.formatMonths((int) value));
```
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/type/IntervalYearMonthOperators.java]: Use `Ints.checkedCast`.

**Comment by haozhun** [comment]
@martint @dain This pull request is simple and clean. It seems good to me. It doesn't feel like this would have any backward compatibility problems. I would also like to understand what problem the issue #5434 is trying to address.

**Comment by haozhun** [comment]
@wrmsr Can you address the simple comment I have, and rebase your change on top of latest trunk?

**Comment by wrmsr** [comment]
@haozhun Done =]

**Comment by haozhun** [comment]
Merged #5716. Thank you.

I changed the commit message and squashed the commits.

**Pull request closed without merging**